### PR TITLE
chore: update to latest ecosystem packages

### PR DESCRIPTION
**Pull request by tim-smart**

### CODE REVIEW

**Comment by tim-smart** [comment]
I had a look at what the dual API migration would take, and I'm not sure if is a good idea with all the `options?: AnnotationOptions`. This would require variants for each constructor.

**Comment by gcanti** [comment]
I also think it's not a good idea, I would leave the APIs as they are.

File: test/compiler/TypeScript.ts [hunk 1]
```diff
@@ -168,9 +174,11 @@ const typeScriptFor = <A>(schema: S.Schema<A>): TypeScript<A> => {
         const id = pipe(
           getIdentifier(ast),
           O.map((id) => ts.factory.createIdentifier(id)),
-          O.getOrThrow(() =>
-            new Error(`cannot find an indentifier for this unique symbol ${String(ast.symbol)}`)
-          )
+          O.getOrElse(() => {
```
Comment (gcanti) [inline test/compiler/TypeScript.ts:177]: I released `@fp-ts/core@0.2.1` which contains `O.getOrThrowWith`.

Comment (tim-smart) [reply test/compiler/TypeScript.ts]: Nice :)

**Comment by mikearnaldi** [comment]
I'll jump ahead and merge this for now as I need it, we can decide later on the api choices such as dual

**Pull request merged**